src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -137,6 +137,9 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                 {
                     await RestoreAsync(request.ForceRestore, request.RestoreSource, token);
                 }
+                catch (OperationCanceledException) when (token.IsCancellationRequested)
```

Comment:
OK, if this isn't a public API that outsiders can call, I guess that makes sense. But this doesn't appear to be the top-level method. So IMO the new catch should be in the top-level caller rather than this method, that way you can still suppress the activitylog entry, but maintain the typical contract of throwing when cancelling from your library-level method.